ext/AtomixMetalExt.jl

```diff
@@ -30,6 +30,21 @@ end
     return (; old = old, success = old === expected)
 end
 
+
+# CAS is needed for FP ops on ThreadGroup memory
+@inline function Atomix.modify!(ref::IndexableRef{<:MtlDeviceArray{<:AbstractFloat, <:Any, Metal.AS.ThreadGroup}} , op::OP, x, order) where {OP}
+    x = convert(eltype(ref), x)
+    ptr = Atomix.pointer(ref)
+    begin
+        old = if op === (+) || op === (-)
+            Metal.atomic_fetch_op_explicit(ptr, op, x)
+        else
+            error("not implemented")
+        end
```

Comment:
Why limit this to `+` and `-`? `atomic_fetch_op_explicit` is generic, that's the whole purpose of the CAS loop.

```suggestion
        Metal.atomic_fetch_op_explicit(ptr, op, x)
```

In addition, we should probably use it as the fallback of `modify!` instead of `error`ing out (the same change could be made to the CUDA extension).

Reply:
it's limited since I haven't checked if other operations need CAS or not